### DISCUSSION TITLE
"package" as a query parameter name

### DISCUSSION THREAD

**Discussion by terencemcdevitt-argondigital** [Netflix/dgs-framework#1306] [General]
We're using DGS to build a Java client library that interfaces with a third party GraphQL API. In a number of query definitions in their schema, they're using "package" as the parameter name (I know... :( ).
It looks like DGS is converting this to "_package" - I assume because package is a reserved word in Java?
Anyway, when we use that param in our query builder then execute the query, the API complains because it doesn't know a param named "_package".
What options do we have here? We can't (easily) get the third party to change the param name.
Are we able to change the name of the parameter once it's been added to the query builder?
Thanks

**Comment by srinivasankavitha** [comment]
Yes keywords are converted to _keyword in codegeneration. What is the error you are seeing? Is it that it is adding `_package` as the name in the query string instead of `package`? That seems like a bug that needs to be fixed in codegen.

**Comment by terencemcdevitt-argondigital** [comment]
Hey @srinivasankavitha - that's correct. When we actually set that parameter on the query, the name that's being used includes the underscore. As a result, the API complains because it doesn't recognize that parameter name.
In the current release, is there some way for us to change that param name at the time we're building the query? It would be a workaround, for sure, but would at least get us moving forward.

**Comment by srinivasankavitha** [comment]
Unfortunately, there isn't. The way you can work around is to just use a manual query string to pass in instead of the API builder generated by codegen.

**Comment by terencemcdevitt-argondigital** [comment]
Ok, thanks.
Do you have a list of the reserved words? I'm going to search the entire schema we're working with to see if there are other places this is happening.

**Comment by srinivasankavitha** [comment]
The list of keywords comes from javax.lang.model.SourceVersion:
https://github.com/Netflix/dgs-codegen/blob/master/graphql-dgs-codegen-core/src/main/kotlin/com/netflix/graphql/dgs/codegen/generators/java/ReservedKeywordSanitizer.kt
